code/datums/ai/ai.dm

```diff
@@ -1,5 +1,27 @@
 /datum/ai
 	var/name
+	var/mob/living/body		// The parent mob we control.
+	var/wait_for = 	0 		// The next time we can process.
+	var/run_interval = 1 	// How long to wait between processes.
+	
+/datum/ai/New(var/mob/living/body)
+	src.body = body
 
-/datum/ai/proc/process()
-	return PROCESS_KILL
+/datum/ai/proc/can_process()
+	if((body.client || body.mind) && !(body.status_flags & ENABLE_AI))
+		return FALSE
+	if(wait_for > world.time)
+		return FALSE
+	if(body.stat == DEAD)
+		return FALSE
+	wait_for = world.time + run_interval
+	return TRUE
+
+/datum/ai/proc/do_process()
+	if(!can_process())
+		return FALSE
+	return TRUE
```

Comment:
By having the following setup, things will be easier for subtypes:
```
/datum/ai/proc/process()
    if(can_process()
        do_process()
```
The reason being is that when a subtype overrides `do_process()` one does not have to remember to call `..()`, check the return value, and then return one's own appropriate value.